_source/_posts/2022-01-30-local-https-java.adoc

```diff
@@ -0,0 +1,278 @@
+---
+layout: blog_post
+title: "HTTPS for Local Java Development"
+author: brian-demers
+by: advocate
+communities: [java]
+description: ""
+tags: [java, security, https, localhost, spring, spring-boot]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+If you're developing a web application, chances are you want your dev environment as close to production as possible. One of the often-overlooked differences between local development and production servers is the use of Transport Layer Security (TLS), or Hypertext Transfer Protocol Secure (HTTPS). In this post, I'll cover three different options to get your local Java app running with TLS in no time!
+
+toc::[]
+
+== Start with a Simple Java Application
+
+To test out these options, you will, of course, need a Java web application. If you already have one handy, skip this step!
+
+Bootstrap a new application using the https://start.spring.io/#!type=maven-project&language=java&packaging=jar&jvmVersion=11&groupId=com.example&artifactId=demo&name=demo&description=Demo%20project%20for%20Spring%20Boot&packageName=com.example.demo&dependencies=web,devtools[Spring Initializer]. Clicking on that link will automatically add the `web` and `devtools` dependencies.
```

Comment:
I like the no Spring Boot version number. Since our posts with versions get deprecated and stop working after a couple years, I think we need to switch to no version number of `git clone` as the defaults.